### PR TITLE
Adds Cassandra healthCheck (0.8.x)

### PR DESCRIPTION
**Pull request by ionutioio**
Creates a healthcheck for the Cassandra Database

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -13,6 +13,7 @@ val log4CatsVersion = "1.1.1"
 val http4sVersion = "0.21.22"
 val circeVersion = "0.13.0"
 val sttpVersion = "1.7.2"
+val cassandraVersion ="4.12.0"
```
Comment (kubukoz) [inline build.sbt:16]: File needs a scalafmt run :)

File: modules/cassandra/src/main/scala/sup/cassandra.scala [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import cats.Id
+import cats.effect.implicits.catsEffectSyntaxConcurrent
+import cats.effect.{Concurrent, Sync, Timer}
+import com.datastax.oss.driver.api.core.CqlSession
+
+import scala.concurrent.duration.FiniteDuration
+
+object cassandra {
+
+  def connectionCheck[F[_] : Sync : Concurrent](session: CqlSession, timeout: FiniteDuration)(implicit timer: Timer[F]): HealthCheck[F, Id] =
```
Comment (kubukoz) [inline modules/cassandra/src/main/scala/sup/cassandra.scala:12]: please run scalafmt here as well (I recommend just enabling "format on save" in your editor :) )

Comment (kubukoz) [reply modules/cassandra/src/main/scala/sup/cassandra.scala:12]:
```suggestion
  def connectionCheck[F[_] : Concurrent: Timer](session: CqlSession, timeout: FiniteDuration): HealthCheck[F, Id] =
```

File: modules/cassandra/src/main/scala/sup/cassandra.scala [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    HealthCheck.liftFBoolean {
+      Sync[F].delay(session
+        .execute("SELECT now() FROM system.local").isFullyFetched
+      ).timeoutTo(timeout, Sync[F].pure(false))
```
Comment (kubukoz) [inline modules/cassandra/src/main/scala/sup/cassandra.scala:16]: We can reuse this mod here: https://sup.kubukoz.com/guide/#timeouttosick

**Review by kubukoz — CHANGES_REQUESTED** [review]

**Comment by kubukoz** [comment]
What do you think about doing the timeout separately, i.e. having users wrap the healthcheck in a `timeoutToSick` on their own? The only other module that has a timeout is the doobie one, but that's pretty much the only way we can have timeouts there as far as I know.

File: modules/cassandra/src/main/scala/sup/cassandra.scala (hunk 3 above)
Comment (ionutioio) [reply modules/cassandra/src/main/scala/sup/cassandra.scala]: I changed the way I did it. It'll be the client who will use https://sup.kubukoz.com/guide/#timeouttosick

File: build.sbt (hunk 1 above)
Comment (ionutioio) [reply build.sbt]: I ran scalafmt everywhere :)

File: modules/cassandra/src/main/scala/sup/cassandra.scala (hunk 2 above)
Comment (ionutioio) [reply modules/cassandra/src/main/scala/sup/cassandra.scala]: i actually removed the timer altogether

File: modules/cassandra/src/main/scala/sup/cassandra.scala (hunk 3 above)
Comment (kubukoz) [reply modules/cassandra/src/main/scala/sup/cassandra.scala]: Cool :)

**Comment by kubukoz** [comment]
I'm not sure if GHA gives us docker access out of the box, but since the tests currently don't run on the `0.8.x` branch (not sure why) I'm okay with just merging this 😅 tests pass locally.

FYI I made a small change to the API (made blocker explicit, as that's the general recommendation) and some test cleanup. Note that the `close` part of the session will be much easier in the CE3 part (`IO.fromCompleteableFuture` or something like that)

File: modules/cassandra/src/main/scala/sup/cassandra.scala [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import cats.effect.{Concurrent, Sync, Timer}
+import com.datastax.oss.driver.api.core.CqlSession
+
+import scala.concurrent.duration.FiniteDuration
+
+object cassandra {
+
+  def connectionCheck[F[_] : Sync : Concurrent](session: CqlSession, timeout: FiniteDuration)(implicit timer: Timer[F]): HealthCheck[F, Id] =
+    HealthCheck.liftFBoolean {
+      Sync[F].delay(session
```
[the remark this answers is not in the document]
Comment (ionutioio) [reply modules/cassandra/src/main/scala/sup/cassandra.scala]: Fixed :)

**Comment by ionutioio** [comment]
Thank you for your help :) will get to the other mr ASAP. Will be polling mvn central for the new version :)

**Comment by kubukoz** [comment]
Seems to have worked after some attempts :) https://repo1.maven.org/maven2/com/kubukoz/sup-cassandra_2.13/0.8.1/

**Pull request merged**